pkg/artifactory/datasource/repository/federated/datasource_artifactory_federated_generic_repository.go

```diff
@@ -0,0 +1,55 @@
+package federated
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/jfrog/terraform-provider-artifactory/v7/pkg/artifactory/datasource/repository"
+	resource_repository "github.com/jfrog/terraform-provider-artifactory/v7/pkg/artifactory/resource/repository"
+	"github.com/jfrog/terraform-provider-artifactory/v7/pkg/artifactory/resource/repository/federated"
+	"github.com/jfrog/terraform-provider-artifactory/v7/pkg/artifactory/resource/repository/local"
+	"github.com/jfrog/terraform-provider-shared/packer"
+	"github.com/jfrog/terraform-provider-shared/predicate"
+	"github.com/jfrog/terraform-provider-shared/util"
+)
+
+func DataSourceArtifactoryFederatedGenericRepository(repoType string) *schema.Resource {
```

Comment:
I know we use `repoType` in many places but it's actually `packageType`. I used that term in remote repo data source so let's use it here too. And we can slowly migrate existing variable names in the future.